src/content/en/updates/2017/03/performant-expand-and-collapse.md

```diff
@@ -0,0 +1,294 @@
+{# wf_featured_snippet: When making expand and collapse effects you can use scale transforms to keep your UI performant. #}
+
+
+# Building performant expand & collapse animations {: .page-title }
+
+{% include "web/_shared/contributors/paullewis.html" %}
+{% include "web/_shared/contributors/smcgruer.html" %}
+
+**TL;DR - use scale transforms when animating clips. You can prevent the children from being
+stretched and skewed during the animation by counter-scaling them.**
+
+Previously we’ve posted updates on how to create performant 
+[parallax effects](/web/updates/2016/12/performant-parallaxing) and 
+[infinite scrollers](/web/updates/2016/07/infinite-scroller). In this 
+post we’re going to look over what’s involved if you want performant clip animations. If you want 
+to see a demo, check out the [Sample UI Elements GitHub repo](https://github.com/GoogleChrome/ui-element-samples/tree/gh-pages/animated-clip).
+
+Take, for example, an expanding menu:
+
+<div style="display: flex; justify-content: center">
+  <video src="/web/updates/images/2017/03/performant-expand-and-collapse/menu.mp4" loop autoplay>
+  </video>
+</div>
+
+Some options for building this are more performant than others.
+
+## Bad: Animating width and height on a container element
+
+You could imagine using a bit of CSS to animate the width and height on the container element.
+
+    .menu {
+      overflow: hidden;
+      width: 350px;
+      height: 600px;
+      transition: width 600ms ease-out, height 600ms ease-out;
+    }
+
+    .menu--collapsed {
+      width: 200px;
+      height: 60px;
+    }
+
+The immediate problem with this approach is that it requires animating `width` and `height`. 
+These properties require calculating layout and paint the results on every frame of the animation, 
+which can be very expensive, and will typically cause you to miss out on 60fps. If that’s news to 
+you then read our 
+[Google Web Fundamentals Rendering Performance](/web/fundamentals/performance/rendering/)
+guides, where you can get more information on how the rendering process works.
+
+## Bad: Use the CSS clip or clip-path properties.
+
+An alternative to animating `width` and `height` might be to use the (now-deprecated) `clip` 
+property to animate the expand and collapse effect. Or, if you prefer, you could use `clip-path` 
+instead. Using `clip-path`, however, is [less well supported](http://caniuse.com/css-clip-path) 
+than `clip`. But `clip` is deprecated. Right. But don’t despair, this isn’t the solution you wanted 
+anyway!
+
+    .menu {
+      position: absolute;
+      clip: rect(0px 112px 175px 0px);
+      transition: clip 600ms ease-out;
+    }
+
+    .menu--collapsed {
+      clip: rect(0px 70px 34px 0px);
+    }
+
+While better than animating the `width` and `height` of the menu element, the downside of this 
+approach is that it still triggers paint. Also the `clip` property, should you go that route, 
```

Comment:
(Side Question: Is there a way to write a general article on how to avoid triggering paint? Or do I need to learn all the places it happens to effectively avoid it? This is a common theme that keeps coming up. It would be nice to teach people how to fish, rather than giving them fish.)